packages/reporting-server/setup.py

```diff
@@ -22,6 +22,7 @@
         "tortoise-orm~=0.16.21",
         "pyjwt[crypto]~=2.0",
         "pydantic~=1.8",
+        "aerich~=0.5.3",
```

Comment:
I'm not sure if I got this right, aerich is only ran as a pre-startup script, so technically, the server doesn't really need aerich as part of the application?

Or does the server run aerich during it's startup process?

If it is the former, perhaps aerich should be a dev dependency. On 2nd thought, if we use the latter, would it be possible to do the one container deployment? Since aerich is ran as part of the startup process, the database must be already up. We can either use kubernetes health probe to wait for the db to be started before starting the server pod, or simply have the startup process try connecting to the database several times with a timeout.

Reply:
> I'm not sure if I got this right, aerich is only ran as a pre-startup script, so technically, the server doesn't really need aerich as part of the application?

Yes, you're right, just to check I should put this in the `extras_require`? 

```
    extras_require={
        "postgres": ["asyncpg~=0.22.0"],
        "mysql": ["aiomysql~=0.0.21"],
        "maria": ["aiomysql~=0.0.21"],
        "aerich": ["aerich~=0.5.3"],
    }
```

Reply:
Hmm, I think this is pretty complicated. I understand the dependencies as the packages required to run the application. In this case, since aerich is technically not part of the application, you can `pip install api-server` and run `rmf-api-server` without aerich installed. But however, aerich is needed to do a proper deployment with migrations.

I think if we can build aerich into the application, then this problem can be solved, along with the database and migration startup race condition.

Reply:
So you mean to run it wvwry time before the reporting-server starts?

Reply:
I think in the worst case, you can spawn a sub-process and wait for it to complete before proceeding with the rest of the init phase.

Reply:
We can do this for development, but it's not the best practice to do it on deployment.  Every instance of the application will attempt to migrate the database. The application has permission to perform destructive updates to the database (this can change in the future). Unless we have only one instance deployed. 

Here is an interesting article related to that https://andrewlock.net/deploying-asp-net-core-applications-to-kubernetes-part-7-running-database-migrations/

Reply:
Thanks, thats a great article. I guess kubernetes deployment is still quite new and there isn't an established best practice yet, another web app I worked on had database migrations builtin but I guess it also detects if a migration is ongoing so only 1 instance of the app performs it and the others wait.

There is one part I'm not quite sure on the example, it starts performing the migration while v1 is still running, wouldn't it cause the transactions to fail? Seems like what should happen is that we deploy v2 before the migration, and v2 will have to support both the old and new schema, it will try the new schema first and if it fails it will fallback to the old schema. Then when the migration is complete we can deploy a v3 with the old schema handlers removed.

If we follow this convention (migration in an init container), is it still possible to support in memory db? We can't really migrate an in memory db before starting the app. I guess this also makes it fundamentally different from a traditional app where one would be able to start it by running an executable.

@codebot Would this compromise be acceptable? Or should we make the extra effort to still let it work with in memory db and single executable by implementing some intelligent migrations detection logic into the app?